documentation/development-docs/patterns-for-standard-cmdlets.md

```diff
@@ -0,0 +1,996 @@
+        public string Name { get; set; }
+
+        [Parameter(Mandatory = true, ValueFromPipeline = true, ParameterSetName = "UpdateByInputObjectParameterSet")]
+        [ValidateNotNull]
+        public PSFoo InputObject { get; set; }
+
+        [Parameter(Mandatory = true, ValueFromPipelineByPropertyName = true, ParameterSetName = "UpdateByResourceIdParameterSet")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+
+        [Parameter(Mandatory = false)]
+        public Type1 Property1 { get; set; }
+
+        [Parameter(Mandatory = false)]
+        public Type2 Property2 { get; set; }
+
+        // Excluding other property parameters
+
+        public override void ExecuteCmdlet()
+        {
+            if (this.IsParameterBound(c => c.InputObject))
+            {
+                this.ResourceGroupName = this.InputObject.ResourceGroupName;
+                this.Name = this.InputObject.Name;
+            }
+
+            if (this.IsParameterBound(c => c.ResourceId))
+            {
+                var resourceIdentifier = new ResourceIdentifier(this.ResourceId);
+                this.ResourceGroupName = resourceIdentifier.ResourceGroupName;
+                this.Name = resourceIdentifier.ResourceName;
+            }
+
+            Foo foo = null;
+            try
+            {
+                foo = this.MySDKClient.Foo.Get(this.ResourceGroupName, this.Name);
+            }
+            catch
+            {
+                foo = null;
+            }
+
+            if (foo == null)
+            {
+                throw new Exception(string.Format("A Foo with name '{0}' in resource group '{1}' does not exist. Please use New-AzureRmFoo to create a Foo with these properties.", this.Name, this.ResourceGroupName));
+            }
+
+            foo.Property1 = this.IsParameterBound(c => c.Property1) ? this.Property1 : foo.Property1;
+            foo.Property2 = this.IsParameterBound(c => c.Property2) ? this.Property2 : foo.Property2;
+            ...
+
+            if (this.ShouldProcess(this.Name, string.Format("Updating Foo '{0}' in resource group '{1}'.", this.Name, this.ResourceGroupName)))
+            {
+                var result = new PSFoo(this.MySDKClient.Foo.Update(this.ResourceGroupName, this.Name, foo)
+                WriteObject(result);
+            }
+        }
+    }
+}
+```
+
+</p>
+</details>
+
+## Child Resources
+
+### `Get-*` cmdlet
+
+All child resources should have a `Get-*` cmdlet that allows users to list the child resources under a parent resource, as well as get a specific resource. In addition, users should be able to provide the resource id of the resource they want to get, and the cmdlet will parse the string to get the necessary identity information.
+
+#### Parameter sets
+
+To enable the scenarios mentioned previously, the cmdlet will need three parameter sets:
+
+```
+Get-AzureRmChildFoo -ResourceGroupName <String> -ParentName <String> [-Name <String>]
```

Comment:
In this case, ParentName should be resource specific, i.e VMName ot NetworkName for sub-resources of VM and Network respectively. In this case, FooName

Reply:
Updated across child resource cmdlets.